### ISSUE TITLE
/usr/bin/env: ‘node’: No such file or directory

### ISSUE THREAD

**Issue by Jamesking56** [mbolotov/intellij-cypress#101]
When I run Cypress inside of PHPStorm to run all tests inside of a directory, I get:

```
/usr/bin/env: ‘node’: No such file or directory
```

Node is definitely present, as cypress runs fine normally in the terminal. I've checked the run configuration and it all points to the correct Node paths...

Any ideas what else I can check here?

I'm running PHPStorm 2022.2 on PopOS 22.04

**Comment by mbolotov** [comment]
Hi James,
Do you mean that the problem is only present for the 'all tests inside of a directory' mode?
Can you post the entire first line of the console output (the one with start command)?

**Comment by Jamesking56** [comment]
@mbolotov Yes exactly, here is the output in PHPStorm's test window (had to censor the project name as its confidential):

```
/home/jamesking56/.nvm/versions/node/v16.16.0/bin/npx cypress run --reporter /tmp/intellij-cypress-reporter12805111157771413441.js --spec /home/jamesking56/PhpstormProjects/{censored}/tests/cypress/**/*
/usr/bin/env: ‘node’: No such file or directory
```

**Comment by mbolotov** [comment]
can you also provide the command for a single test which is working okay?
what's the difference?

**Comment by Jamesking56** [comment]
Single tests aren't working OK and are showing the same error

**Comment by mbolotov** [comment]
It's a bit hard to reproduce the problem as I'm on Windows mostly of the time. 
So you mean when you run this command directly in terminal, it working okay?

```
cd /home/jamesking56/PhpstormProjects
npx cypress run  --spec /home/jamesking56/PhpstormProjects/{censored}/tests/cypress/**/*
```

Also, you could try creating a symlink as described here: https://stackoverflow.com/questions/26320901/cannot-install-nodejs-usr-bin-env-node-no-such-file-or-directory

**Comment by Jamesking56** [comment]
Yes that command works in the terminal as normal and I've added the symlink and it does not fix the issue

**Comment by mbolotov** [comment]
this might be related to the user under you run PHPStorm itself. 
is it the same user as the one use in terminal?

**Comment by Jamesking56** [comment]
Yes it is the same user, installed via Jetbrains Toolbox

**Comment by mbolotov** [comment]
you can also play with the settings on `runner` tab:
![image](https://user-images.githubusercontent.com/19785715/185107123-535315ed-7282-4cdb-bf36-bd5cfc6fee6b.png)
if you clear the `Package Manager` field, you'll be able to select and use a NodeJS interpreter directly.

**Comment by Threnos** [comment]
Today I faced this as well. It was working for quite a while until today when I updated node via nvm from 16.14.2 to 16.17.0.
Clicking "Run all tests in cypress" opens runner window where it is clearly seen that old node version is chosen (that was deleted after an update)
![image](https://user-images.githubusercontent.com/30055497/187666369-ed2d40d1-f408-44a7-8a67-21cb52eb3b00.png)
Everything else works with new node as expected without any issues.

**Comment by list4c** [comment]
I have the same issue, also using nvm and linux. 
Tried running a single test but it failed.

**Comment by jippeholwerda** [comment]
This is due to the fact that from where PHPStorm/Webstorm is started, the correct terminal config / env vars are not present. When starting PHPStorm/Webstorm from the terminal it probably works.
See https://youtrack.jetbrains.com/issue/IDEABKL-7589
